### PR TITLE
fix:#19 add support to generate names randomly

### PR DESCRIPTION
**Pull request by erdahuja**

### CODE REVIEW

File: __tests__/unique-names-generator.test.ts [hunk 1]
```diff
@@ -185,4 +196,26 @@ describe('randomNameGenerator', () => {
...
+      separator: '_',
+      random: true,
+    };
+
+    // Act
+    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
+    const expected = () => uniqueNamesGenerator.generate();
+
+    // Assert
+    // expect(() => expected()).toThrowErrorMatchingSnapshot();
```
Comment (erdahuja) [inline __tests__/unique-names-generator.test.ts]: @andreasonny83 Can you help me out here please. how would i test it?

Comment (andreasonny83) [reply __tests__/unique-names-generator.test.ts]: Hey @erdahuja . Can you please rename `random` to `shuffleDictionaries` instead.
Also, please don't commit the `yarn.lock`.

File: __tests__/unique-names-generator.test.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ describe('randomNameGenerator', () => {
       dictionaries: [[], [], []],
       separator: '_',
       length: 3,
+      random: false,
```
Comment (andreasonny83) [inline __tests__/unique-names-generator.test.ts]: rename to `shuffleDictionaries`

File: lib/unique-names-generator.ts [hunk 3]
```diff
@@ -7,19 +7,24 @@ export interface Config {
   separator: string;
   length: number;
   dictionaries: string[][];
+  random: boolean,
```
Comment (andreasonny83) [inline lib/unique-names-generator.ts]: rename to `shuffleDictionaries`

File: lib/unique-names-generator.ts [hunk 4]
```diff
@@ -7,19 +7,24 @@ export interface Config {
...
 
 export class UniqueNamesGenerator {
   private dictionaries: string[][];
   private length: number;
   private separator: string;
-
+  private random: boolean;
+  private generateRandomNumber(low: number = 0, high: number) {
+    return Math.floor(Math.random()*(high - low)) + low;
+  }
```
Comment (andreasonny83) [inline lib/unique-names-generator.ts]: move this private method at the bottom of the class after the `generate` public one

File: lib/unique-names-generator.ts [hunk 5]
```diff
@@ -42,9 +47,20 @@ export class UniqueNamesGenerator {
...
+        const randomDictionary = this.dictionaries[this.generateRandomNumber(0, this.length)];
+        randomDictionaries = [...randomDictionaries, randomDictionary];
+      }
+      dictionariesToUse = randomDictionaries;
+    } else {
+      dictionariesToUse = this.dictionaries.slice(0, this.length);
+    }
+    return dictionariesToUse.reduce((acc: string, curr: string[]) => {
+      const rnd = this.generateRandomNumber(0, curr.length)
+      console.log(curr, rnd)
```
Comment (andreasonny83) [inline lib/unique-names-generator.ts]: remove `console.log`

**Comment by andreasonny83** [comment]
Also, I just noticed the PR is against an older version of the library. Do you mind if I take your suggestion and I implement that directly on my side?

File: lib/unique-names-generator.ts [hunk 6]
```diff
@@ -42,9 +47,20 @@ export class UniqueNamesGenerator {
           }`,
       );
     }
-
-    return this.dictionaries.slice(0, this.length).reduce((acc: string, curr: string[]) => {
-      const rnd = Math.floor(Math.random() * curr.length);
+    let dictionariesToUse: string[][];
+    if (this.random) {
+      let randomDictionaries: string[][] = [];
```
Comment (andreasonny83) [inline lib/unique-names-generator.ts]: or just use the Array sort method
```
    const dictionariesToUse = this.dictionaries.sort(() => {
      if (this.shuffleDictionaries) {
        return this.generateRandomNumber(0, this.length);
      }

      return 0;
    });
```

File: __tests__/unique-names-generator.test.ts (hunk 1 above)
Comment (andreasonny83) [reply __tests__/unique-names-generator.test.ts]: I would probably test the logic in this way:
```
  it('should always use the same dictionaries when shuffleDictionaries is disabled', () => {
    // Arrange
    const dictionary1 = ['dictionary1'];
    const dictionary2 = ['dictionary2'];
    const dictionary3 = ['dictionary3'];
    const dictionary4 = ['dictionary4'];
    const dictionary5 = ['dictionary5'];

    const config: Config = {
      dictionaries: [dictionary1, dictionary2, dictionary3, dictionary4, dictionary5],
      length: 2,
      separator: '_',
      shuffleDictionaries: false,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    const res1 = uniqueNamesGenerator.generate();
    const res2 = uniqueNamesGenerator.generate();
    const res3 = uniqueNamesGenerator.generate();

    // Assert
    expect(res1).toEqual(res2);
    expect(res1).toEqual(res3);
  });

  it('should shuffle the provided dictionaries when shuffleDictionaries is enabled', () => {
    // Arrange
    const dictionary1 = ['dictionary1'];
    const dictionary2 = ['dictionary2'];
    const dictionary3 = ['dictionary3'];
    const dictionary4 = ['dictionary4'];
    const dictionary5 = ['dictionary5'];

    const config: Config = {
      dictionaries: [dictionary1, dictionary2, dictionary3, dictionary4, dictionary5],
      length: 2,
      separator: '_',
      shuffleDictionaries: true,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    const res1 = uniqueNamesGenerator.generate();
    const res2 = uniqueNamesGenerator.generate();

    // Assert
    expect(res1).not.toEqual(res2);
  });
```

**Comment by erdahuja** [comment]
@andreasonny83 done suggested changes, Once review complete i will create a fresh PR with updated library and rebased commit

File: lib/unique-names-generator.ts (hunk 6 above)
Comment (erdahuja) [reply lib/unique-names-generator.ts]: Great idea ! 👯‍♂

**Comment by andreasonny83** [comment]
Closing this as outdated. Feel free to reopen if you want to update that

**Pull request closed without merging**